clusterloader2/docs/GETTING_STARTED.md

```diff
@@ -0,0 +1,303 @@
+
+Check that you can connect to cluster:
+```bash
+kubectl get nodes
+```
+
+
+## Prepare simple test to run
+Let's prepare our first test config (config.yaml).
+This test will:
+- Create one namespace
+- Create a single deployment with 10 pods inside that namespace
+- Measure startup latency of these pod
+
+We will create file `config.yaml` that describes this test.
+First we need to start with defining test name:
+```yaml
+name: test
+```
+CL2 will create namespaces automatically, but we need to specify how many namespaces we want:
+```yaml
+namespace:
+  number: 1
+```
+Next, we need to specify TuningSets.
+TuningSet describes how actions are executed.
+In our case, we will have only 1 deployment so there will be only 1 action to execute.
+```yaml
+tuningSets:
+- name: Uniform1qps
+  qpsLoad:
+    qps: 1
+```
+Test definition consists of a list of steps. A step can be either collection of Phases or Measurements.
+A Phase defines a state the cluster should reach.
+A Measurement allows to measure something or wait for something.
+You can find list of available measurements here [Measurements].
+Our first step will be starting two measurements.
+We want to start measuring pod startup latency and also make measurement that will wait for all pods to be in running state.
+Setting the action field to `start` begins execution of measurement.
+For both measurements we need to specify labelSelectors so they know which pods they should take into account.
+PodStartupLatency also takes threshold. If 99th percentile of latency will go over this threshold, test will fail.
+```yaml
+steps:
+- name: Start measurements
+  measurements:
+  - Identifier: PodStartupLatency
+    Method: PodStartupLatency
+    Params:
+      action: start
+      labelSelector: group = test-pod
+      threshold: 20s
+  - Identifier: WaitForControlledPodsRunning
+    Method: WaitForControlledPodsRunning
+    Params:
+      action: start
+      apiVersion: apps/v1
+      kind: Deployment
+      labelSelector: group = test-pod
+      operationTimeout: 120s
+```
+Once we created these two measurements, we can have next step that creates deployment.
+We need to specify in which namespaces we want this deployment to be created, how many of these deployments per namespace.
+Also, we will need to specify template for our deployment, which we will do later.
+For now, let's assume that this template allows us to specify number of replicas in deployment.
+```yaml
+- name: Create deployment
+  phases:
+  - namespaceRange:
+      min: 1
+      max: 1
+    replicasPerNamespace: 1
+    tuningSet: Uniform1qps
+    objectBundle:
+    - basename: test-deployment
+      objectTemplatePath: "deployment.yaml"
+      templateFillMap:
+        Replicas: 10
+```
+Now, we need to wait for pods in this deployment to be in Running state before we measure PodStartupLatency.
```

Comment:
To be precise - measuring of PodStartupTime i happening in the background.
It's only aggregation that is happening later - so we need to wait for these pods to be started to have all data to be aggregated.